apps/voting/app/src/hooks/useVotes.js

```diff
@@ -58,7 +59,14 @@ function useDecoratedVotes() {
         executionTargetData = {
           address: appAddress,
           name,
-          iconSrc: icon(24),
+
+          // This check ensures icon() only gets called if it actually exists,
+          // as users were reporting a crash happening in some cases. The
+          // problem seems to come from the multiple targets condition that
+          // was not including icon(), but this ensures it won’t crash anymore,
+          // until we can 100% reproduce the issue.
```

Comment:
Maybe enforce a shape on `targetApp`?
```
const TARGET_APP_SHAPE = {
  icon: () => null,
  identifier: '...'
}
targetApp = {
...TARGET_APP_SHAPE,
...targetApp
}
```

Reply:
And I was able to reproduce consistently: using Safari, no Frame, reload go to Voting -> Finance -> Voting, it happened every time.

Reply:
Oh cool, could you check if [this alone](https://github.com/aragon/aragon-apps/pull/1012/files#diff-e7b7e89ee1e9dda895d3987bea4d8532R39) solves the issue, so we don’t have to check for `icon()` to be a function?

Reply:
It does, that's the way I was fixing the problem until I saw this PR and that's why I suggested it too, there are 3 scenarios here, so instead of manually setting things in each case we can use the enforce object interface once :)